src/modules/activities/profile/partials/ProfileForm.tsx

```diff
@@ -0,0 +1,225 @@
+    displayName: profile.displayName,
+    firstName: profile.firstName,
+    lastName: profile.lastName,
+    gender: profile.gender,
+    address: profile.address || {},
+    bank: profile.bank || {},
+  });
+  const [isOpen, setIsOpen] = useState(false);
+  const [step, setStep] = useState(1);
+  const onClose = () => {
+    if (isUpdating) return;
+    setIsOpen(false);
+  };
+
+  const togglePopup = () => {
+    setIsOpen(!isOpen);
+  };
+
+  const renderSteps = () => {
+    switch (step) {
+      case 1:
+        return (
+          <PersonalInformation register={register} errors={errors} />
+        );
+      case 2:
+        return (
+          <AddressInformation register={register} errors={errors} />
+        );
+      case 3:
+        return (
+          <BankInformation register={register} errors={errors} />
+        );
+      default:
+        return null;
+    }
+  };
+
+  const updateUserProfile = async (payload: any) => {
+    await dispatch(updateProfile(payload));
+    if (!updateError) {
+      onClose();
+      toast('Profile updated successfully');
+      setStep(1);
+    } else {
+      toast.error(updateError || 'Profile update failed');
+    }
+  };
+
+  const onSubmit = (query: any) => {
+    if (step < 3) {
+      setStep(step + 1);
+    }
+
+    switch (step) {
+      case 1:
+        setProfileData({
+          ...profileData,
+          ...query,
+        });
+        break;
+      case 2:
+        setProfileData({
+          ...profileData,
+          address: {
+            ...profileData.address,
+            ...query,
+          },
+        });
+        break;
+      case 3:
+        setProfileData(prev => {
+          const payload = {
+            ...prev,
+            bank: {
+              ...prev.bank,
+              ...query,
+            },
+          };
+          updateUserProfile(payload);
+          return payload;
+        });
+
+        break;
+      default:
+        break;
+    }
+  };
+
+  const handleBack = () => {
+    if (step > 1) {
+      setStep(step - 1);
+    } else {
+      onClose();
+    }
+  };
+
+  useEffect(() => {
+    const { address, bank, ...others } = profileData;
+    switch (step) {
+      case 1:
+        setSchema(profileSchema.personal);
+        reset(others);
+        break;
+      case 2:
+        setSchema(profileSchema.address);
+        reset(address);
+        break;
+      case 3:
+        setSchema(profileSchema.bank);
+        reset(bank);
+        break;
+      default:
+        break;
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [step]);
+
+  return (
+    <>
+      <Button gradientDuoTone="cyanToBlue" onClick={togglePopup} pill>
+        <HiPencil className="mr-2 h-5 w-5" />
+        Edit
+      </Button>
+      <Modal
+        size="4xl"
+        show={isOpen}
+        position="top-center"
+        onClose={onClose}
+      >
+        <Modal.Header>Edit Profile</Modal.Header>
+        <form
```

Comment:
the best practice when submitting a form is
```
const { handleSubmit } = useForm<FormType>({ ... })
const submitHandler = handleSubmit((data)=>{
      // code here
})
<form onSubmit={submitHandler}  />
```

Reply:
Awesome.